cristin-person/src/main/java/no/unit/nva/cristin/person/employment/update/UpdatePersonEmploymentValidator.java

```diff
@@ -0,0 +1,79 @@
+package no.unit.nva.cristin.person.employment.update;
+
+import static java.util.Objects.isNull;
+import static no.unit.nva.cristin.common.ErrorMessages.ERROR_MESSAGE_INVALID_PAYLOAD;
+import static no.unit.nva.cristin.common.ErrorMessages.invalidFieldParameterMessage;
+import static no.unit.nva.cristin.model.JsonPropertyNames.END_DATE;
+import static no.unit.nva.cristin.model.JsonPropertyNames.ORGANIZATION;
+import static no.unit.nva.cristin.model.JsonPropertyNames.START_DATE;
+import static no.unit.nva.cristin.model.JsonPropertyNames.TYPE;
+import static no.unit.nva.cristin.person.model.nva.JsonPropertyNames.FULL_TIME_PERCENTAGE;
+import static nva.commons.core.attempt.Try.attempt;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.net.URI;
+import java.time.Instant;
+import java.util.Optional;
+import no.unit.nva.cristin.model.CristinOrganization;
+import no.unit.nva.cristin.person.model.nva.Employment;
+import no.unit.nva.utils.UriUtils;
+import nva.commons.apigateway.exceptions.BadRequestException;
+
+public class UpdatePersonEmploymentValidator {
+
+    /**
+     * Validate according to rules of upstream json schema.
+     */
+    public static void validate(ObjectNode input) throws BadRequestException {
+        validateNotNull(input);
+        validatePositionCode(input);
+        validateAffiliation(input);
+        validateDate(input, START_DATE);
+        validateDate(input, END_DATE);
+        validateFullTimePercentage(input);
+    }
+
+    private static void validateNotNull(ObjectNode input) throws BadRequestException {
+        if (isNull(input)) {
+            throw new BadRequestException(ERROR_MESSAGE_INVALID_PAYLOAD);
+        }
+    }
+
+    private static void validatePositionCode(ObjectNode input) throws BadRequestException {
+        if (input.has(TYPE)) {
+            URI type = parseUriField(input, TYPE);
+            Optional<String> code = Employment.extractPositionCodeFromTypeUri(type);
+            if (code.isEmpty() || code.get().isBlank()) {
+                throw new BadRequestException(invalidFieldParameterMessage(TYPE));
+            }
+        }
```

Comment:
The intention here is to accept a type if and only if it has content? So there are two valid states: no type and a populated type?

Reply:
Cristin does not accept null for this field so if we send a value in this field it has to be populated as well. We cannot accept null or empty